packages/authentication/src/types/services/auth-provider.ts

```diff
@@ -1,3 +1,20 @@
+import { AbstractService } from "@medusajs/utils"
+import { IAuthProviderRepository } from "../repositories"
+import { AuthProvider } from "@models"
+
+export interface IAuthProviderService<
+  TEntity extends AuthProvider = AuthProvider
```

Comment:
question: What is the reason behind keeping these service interfaces?

Same question for the repositories that live in `packages/types/cart/repositories.ts` (from a previous PR).

Reply:
Sorry for the repeat. Seb and I was just discussing 😅

Reply:
Because the service does not exists so we can't use it to type it in the module service same for the other things

Reply:
Would it be possible to create the service interface using just the model or is there anything specific we need?

```ts
export interface IAuthProviderService<
  TEntity extends AuthProvider = AuthProvider
> extends AbstractService<
    TEntity,
    {
      authProviderRepository: IAuthProviderRepository<TEntity>
    },
    {
      create: CreateAuthProviderDTO
      update: UpdateAuthProviderDTO
    }
  > {}
```

Reply:
Based on our discussion, either use a base interface that does not enforce typings of the internal services or find a way to have a base typings based on TEntity (second step). same goes for repositories